jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/AddDynamicProcessCommand.java

```diff
@@ -38,34 +39,51 @@
     private String processId;
     private long processInstanceId;
     private Map<String, Object> parameters;
-    
-    public AddDynamicProcessCommand(IdentityProvider identityProvider, String caseId, Long processInstanceId, String processId, Map<String, Object> parameters) {        
+
+    public AddDynamicProcessCommand(IdentityProvider identityProvider,
+                                    String caseId,
+                                    Long processInstanceId,
+                                    String processId,
+                                    Map<String, Object> parameters) {
         super(identityProvider);
         this.caseId = caseId;
         this.processInstanceId = processInstanceId;
         this.processId = processId;
         this.parameters = parameters;
-        
+
         if (processInstanceId == null || processId == null) {
             throw new IllegalArgumentException("Mandatory parameters are missing - process instance id and process id");
         }
     }
 
     @Override
     public Long execute(Context context) {
-        KieSession ksession = ((RegistryContext) context).lookup( KieSession.class );
-        
+        KieSession ksession = ((RegistryContext) context).lookup(KieSession.class);
+
         ProcessInstance processInstance = ksession.getProcessInstance(processInstanceId);
         if (processInstance == null) {
             throw new ProcessInstanceNotFoundException("No process instance found with id " + processInstanceId);
         }
         CaseEventSupport caseEventSupport = getCaseEventSupport(context);
-        caseEventSupport.fireBeforeDynamicProcessAdded(caseId, processInstanceId, processId, parameters);
-        
-        long subProcessInstanceId = DynamicUtils.addDynamicSubProcess(processInstance, ksession, processId, parameters);
-        
-        caseEventSupport.fireAfterDynamicProcessAdded(caseId, processInstanceId, processId, parameters, subProcessInstanceId);
+        caseEventSupport.fireBeforeDynamicProcessAdded(caseId,
+                                                       processInstanceId,
+                                                       processId,
+                                                       parameters);
+
+        long subProcessInstanceId = DynamicUtils.addDynamicSubProcess(processInstance,
+                                                                      ksession,
+                                                                      processId,
+                                                                      parameters);
+
+        if (subProcessInstanceId < 0) {
+            throw new ProcessDefinitionNotFoundException("No process definition found with id: " + processId);
```

Comment:
We shoul get the exception (most likely if different type as DynamicUtils is in jbpm-flow so ProcessDefinitionNotFoundException is not available there) here and then wrap it with ProcessDefinitionNotFoundException and rethow it